### ISSUE TITLE
GUI : Design : Create consistent experience for shutdown & reboot operations

### ISSUE THREAD

**Issue by ryanarnell** [ibm-openbmc/dev#1155]
## Stakeholders
**SME**: @gtmills @anoo1 
**Design Researcher**: @nicoleconser 
**UX Designer**:
**FED**:

## Use Case
We need a consistent way to warn users that the Settings in “Server Power Operations” are set to “cold” (if/whenever they are). User should be warned about this hidden setting whenever they click Reboot or Shut Down anywhere in the GUI (ex: when they perform a Firmware update + reboot). 

## InVision Prototype
https://ibm.invisionapp.com/share/QKNZAGECYNF#/319284132_BMC_Reboot

## Design Issue (phosphor-webui)
TBD

## Development Issue 
TBD

## References/Resources
- Slack discussion: https://ibm-systems-power.slack.com/archives/C0Q6TQP5Z/p1567093743024000?thread_ts=1567091678.014800&cid=C0Q6TQP5Z

- feature discovery folder: 
- user research notes*: 
- user research synthesis: 

*This folder is restricted in accordance with GDPR guidelines.

**Comment by ryanarnell** [comment]
2 quick questions, maybe the answer is no but I thought I'd ask anyway. 
1. Do we need to let user know in this page that any saved settings in host page will be into effect after BMC reboot?
2. Do we need to display the next boot image options here? Or at least display that saved image info which will be the running version if anybody reboots the BMC from here. 

![Screen Shot 2019-10-04 at 2 37 04 PM](https://user-images.githubusercontent.com/4976631/66234902-7a305380-e6b4-11e9-8de0-8710f2030b33.png)

**Comment by anoo1** [comment]
I'd say yes to both. I believe @jandraa is addressing those concerns in the updated fw page.

**Comment by jandraa** [comment]
I don't follow- can yall clarify? 

@ryanarnell :
> Do we need to let user know in this page that any saved settings in host page will be into effect after BMC reboot?

- Do you mean, display in the firmware page the settings configured in the host page? What is the reasoning for that? I'm curious if that's valuable information and if customer care to see it since it isn't changed often. 

> Do we need to display that the saved firmware version (if any from firmware page) will be the running version after someone reboots from this place?

- Can you rephrase this? I don't follow

@anoo1 , what part of the updated fw page design are you thinking about that addresses this? If we already have this covered, what other information do we need to show @ryanarnell ?c

**Comment by ryanarnell** [comment]
1. In host page we give them options for next host boot. We also give them options for saving those host boot settings. Not necessarily reboot the host right away. They can just save settings and leave the host boot page. Those saved settings will not take effect until next time the host reboots. 

Now lets say if someone changed host boot settings in power operations page and then came to this page and rebooted BMC. Since rebooting BMC will reboot host as well, all the saved settings will be applied. If someone rebooted because some other reason (lets say system is unstable) and there are saved settings that can put the host in an unwanted state for this action, do we need to notify the user that Rebooting BMC will apply those saved changes in (host)power operations page. 

2. Similar to no 1. I am going to mention the image above for example- 
a. User goes to firmware page.
b. Selects OP940 for next boot. 
c Saves settings. Doesn't reboot BMC for that setting to take effect. 
d. After sometime (maybe someone else) logs into BMC just to reboot BMC.
e. Goes to BMC reboot page and reboots BMC
f. Ends up with a different firmware version not knowing that rebooting BMC caused it 


In both these cases, do we need to inform user there are host settings (no. 1) or a pending firmware change (no 2) will take effect if he reboots BMC?

**Comment by jandraa** [comment]
I wonder how common those scenarios are. From my knowledge of how both of those pages are used (host boot settings and firmware) they are not something you change often so I don't see it necessary to specify all the settings any time they go into effect. 

On the first scenario you described, host boot settings aren't typically changed unless it's setup time (at which point they would be safe settings) or when something is wrong and they are changing those settings to boot in a different way right then and there. 

Similarly, if the next boot image is changed on the firmware page, chances are they are rebooting right there. I could see the next boot image being valuable information in the Reboot BMC page but not as a popup. 

In my opinion, it is sufficient to keep the next boot image information updated on both screens so that if it is changed in the firmware screens it is reflected in the host settings screens and vice-versa. I don't think all the settings need to be displayed all the time; that seems like an overkill.

**Comment by jandraa** [comment]
@anoo1 From the firmware designs you have reviewed, do you think any boot settings information is missing?

**Comment by ryanarnell** [comment]
Yes. I agree on "all the settings need to be displayed all the time" everywhere. That IS an overkill. Also not a popup- I do not think popup is a good option here. 
Glad that you thought ' the next boot image being valuable information in the Reboot BMC page '

**Comment by jandraa** [comment]
A reboot can occur in three different places in the GUI:

1. A Host reboot on the Power Operations page
2. A BMC reboot on the BMC Reboot page
3. Either a host or BMC reboot (on oBMC) or both (on eBMC) on the firmware page

Because the host reboot is already consistent with the GUI, this story focuses on the BMC reboot only. The reboots happening from the firmware page are consistent with the designs presented here and are addressed in story: #420

**Comment by jandraa** [comment]
Resolved by: 

**A Host reboot on the Power Operations page**
- The design displays a toast notification to indicate reboot has started and when it is successfully complete. All buttons are disabled.
<img width="1421" alt="Screen Shot 2019-10-18 at 3 46 44 PM" src="https://user-images.githubusercontent.com/44675611/67127214-804b2780-f1be-11e9-9590-a6f33a2523c4.png">


**A BMC reboot on the BMC Reboot page** 
- The design displays a toast notification to indicate reboot has started and all buttons are disabled. 
<img width="960" alt="Screen Shot 2019-10-18 at 3 55 32 PM" src="https://user-images.githubusercontent.com/44675611/67127834-bd63e980-f1bf-11e9-9f13-ae91166fa11f.png">

**BMC and host reboot on firmware page**
- The design displays a toast notification to indicate reboot has started, all buttons are disabled, and 'Refresh' button is displayed to allow the user to reload the page. 
<img width="728" alt="Screen Shot 2019-10-18 at 4 19 37 PM" src="https://user-images.githubusercontent.com/44675611/67129025-184b1000-f1c3-11e9-984b-0a0b5bb98fb7.png">


**BMC reboot on firmware page**
- The design displays a toast notification to indicate reboot has started, all buttons are disabled, and 'Refresh' button is displayed to allow the user to reload the page. 
<img width="991" alt="Screen Shot 2019-10-18 at 4 25 02 PM" src="https://user-images.githubusercontent.com/44675611/67129299-dec6d480-f1c3-11e9-87f9-60899f695fc9.png">


**Host reboot on firmware page**
- The design displays a toast notification to indicate reboot has started and when it is successfully complete. All buttons are disabled. 
<img width="1658" alt="Screen Shot 2019-10-18 at 4 45 06 PM" src="https://user-images.githubusercontent.com/44675611/67130624-8db8df80-f1c7-11e9-9621-1c45110e6ee8.png">
